### PR TITLE
Support AccountSet Operations in Serialization

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Add support for AccountSet Serialization.

### Context of Change

We'd like to begin to support additional operation types. `Serializer` acts as a shim layer that converts protocol buffers to JSON objects. We can then feed those JSON objects to ripple-binary-codec and get bytes to sign.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Additional serialization is supported. 

## Test Plan

New tests added for CI.

### CODE REVIEW

File: test/XRP/serializer.test.ts [hunk 1]
```diff
@@ -164,6 +174,127 @@ function makeDepositPreauth(
...
+    senderAddress,
+    publicKey,
+  )
+  transaction.setAccountSet(accountSet)
+
+  return transaction
+}
+
+/**
+ * Make an AccountSet protocol buffer fro the given inputs.
```
Comment (sarahkevinking) [inline test/XRP/serializer.test.ts:230]: nit: `for`

Comment (keefertaylor) [reply test/XRP/serializer.test.ts]: Done. I used `from` instead of `for`. LMK if that distinction bugs you.

File: src/XRP/serializer.ts [hunk 2]
```diff
@@ -1,15 +1,28 @@
+/* eslint-disable  max-lines --
+ * gRPC is verbose. Playing code golf wiht this file would decrease clarity for little readability gain.
```
Comment (sarahkevinking) [inline src/XRP/serializer.ts:2]: nit: with

Code golf sounds like a fun game

File: src/XRP/serializer.ts [hunk 3]
```diff
@@ -1,15 +1,28 @@
+/* eslint-disable  max-lines --
```
Comment (tedkalaw) [inline src/XRP/serializer.ts:1]: is `max-lines` too strict?

**Comment by keefertaylor** [comment]
> did we want to have any kind of basically number -> uint8 (ex: for TickSize) checks anywhere? let me know if it's being done somewhere else

Not positive of what you're asking.  Do you mean making sure the number is 3..15? 
Screen Shot 2020-06-24 at 4 41 12 PM

**Comment by stephengu** [comment]
> > did we want to have any kind of basically number -> uint8 (ex: for TickSize) checks anywhere? let me know if it's being done somewhere else
> 
> Not positive of what you're asking. Do you mean making sure the number is 3..15?
> Screen Shot 2020-06-24 at 4 41 12 PM

yes exactly

**Comment by amiecorso** [comment]
I discussed with Keefer and in keeping with how we've handled this type of logic elsewhere, we're leaving sanity checks to the SDK layer that actually constructs or handles the protobufs, instead of the serialization code.

**Pull request merged**